test/routes.test.js

```diff
@@ -0,0 +1,78 @@
+const index = require('..');
+const assume = require('assume');
+const thenify = require('tinythen');
+const request = require('request-promise-native');
+const StatusHandler = require('../status-handler');
+const fixtures = require('./fixtures');
+const { address } = require('./util');
+const rip = require('rip-out');
+
+
+describe('routes', function () {
+  describe('integration', function () {
+    this.timeout(6E4);
+    let status;
+    let app;
+    let spec;
+
+    before(function (done) {
+      index.start({ auth: false }, (err, instance) => {
+        app = instance;
+        status = new StatusHandler({
+          models: app.models
+        });
+        spec = status._transform(fixtures.singleEvent);
+        done(err);
+      });
+    });
+
+    after(async function () {
+      if (!app) return;
+      await app.models.drop();
+      await thenify(app, 'close');
+    });
+
+    beforeEach(async () => {
+      if (!app) return;
+      await status.event(fixtures.singleEvent);
+      await status.queued(fixtures.singleQueued);
+      await status.complete(fixtures.singleComplete);
```

Comment:
All of my comments are more or less about breaking up this `beforeEach` into more granular tests with these befores:

``` js
// Test event submitted
before(async () => {
  if (!app) return;
  await status.event(fixtures.singleEvent);
});

// Test event submitted and queued correctly
before(async () => {
  if (!app) return;
  await status.event(fixtures.singleEvent);
  await status.queued(fixtures.singleQueued);
});

// Test event submitted, queued, and completed correctly
before(async () => {
  if (!app) return;
  await status.event(fixtures.singleEvent);
  await status.queued(fixtures.singleQueued);
  await status.complete(fixtures.singleComplete);
});
```

Reply:
@indexzero understood, i can fetch from the route at these different parts of the lifecycle